app/src/androidTest/java/com/github/polybooks/LoginGoogleTest.kt

```diff
@@ -0,0 +1,44 @@
+package com.github.polybooks
+
+import android.R
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.action.ViewActions.click
+import androidx.test.espresso.intent.Intents
+import androidx.test.espresso.matcher.RootMatchers.withDecorView
+import androidx.test.espresso.matcher.ViewMatchers.*
+import androidx.test.ext.junit.rules.ActivityScenarioRule
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+
+@RunWith(AndroidJUnit4::class)
+
+class LoginGoogleTest {
+
+    @get:Rule
+    val activityRule = ActivityScenarioRule(LoginActivity::class.java)
+
+    @Before
+    fun before() {
+        Intents.init()
+    }
+
+    @After
+    fun after() {
+        Intents.release()
+    }
+
+    @Test
+    fun signInButtonGoogle() {
+        onView(withId(com.github.polybooks.R.id.sign_in_button)).perform(click())
+    }
+
+    @Test
+    fun signOutButtonGoogle() {
+        onView(withId(com.github.polybooks.R.id.tequila_log_button)).perform(click())
+    }
```

Comment:
I'm no espresso expert, but I believe those two tests can never fail as you click on a button, but doesn't assert any kind of result

Reply:
> I'm no espresso expert, but I believe those two tests can never fail as you click on a button, but doesn't assert any kind of result

They can fail if an error or exception is generated by google api, that's the only way to test the google authentification that I found so far.

Reply:
Are you filling in data of user fields if successful? Or are you moving to the next activity?
If yes, those are things you could assert for.
If not, then, I don't see any way to do more than what you're already doing.